pkg/reftracker/ref_tracker.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"sync"
+)
+
+type AppRefTracker struct {
+	lock       sync.Mutex
+	refsToApps map[RefKey]map[string]struct{}
+	appsToRefs map[string]map[RefKey]struct{}
+}
+
+func NewAppRefTracker() *AppRefTracker {
+	return &AppRefTracker{refsToApps: map[RefKey]map[string]struct{}{}, appsToRefs: map[string]map[RefKey]struct{}{}}
+}
+
+func (a AppRefTracker) AddAppForRef(refKey RefKey, appName string) {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	apps := a.refsToApps[refKey]
+	if apps == nil {
+		apps = map[string]struct{}{}
+	}
+
+	appKey := fmt.Sprintf(`%s:%s`, appName, refKey.Namespace())
```

Comment:
can we use refkeys for this as well?

Reply:
Changed to RefKey in general, but left in place as func param since all that is needed here is AppName.